service/league_service/league_service.py

```diff
@@ -179,11 +179,8 @@ async def is_returning_player(self, player_id: PlayerID, rating_type: str) -> bo
                 )
             )
             result = await conn.execute(sql)
-            row = result.fetchone()
-        if row is None or row.subdivision_id is None:
-            return False
-        else:
-            return True
+            rows = result.fetchall()
```

Comment:
What if you added `where subdivision_id <> null` and `limit 1` to the SQL query? Then you're basically getting the info you want straight from the DB rather than having to pull an ever increasingly large list into memory. 

I’m guessing the seasons are long enough that realistically it would never cause any sort of performance or memory problems, but still I feel like if it can be done more efficiently in the SQL there’s not really a downside.

Reply:
I also don't see a downside